tests/unit-tests/test-class-sensei-messages.php

```diff
@@ -100,4 +102,81 @@ public function testUserMessagesCapCheckAsNonParticipant() {
 		$this->login_as_admin();
 		$this->assertEquals( [ 'read' => true ], $instance->user_messages_cap_check( [], [ 'read' ], [ 'read_post', get_current_user_id(), $message_id ] ), 'Admins should still have access' );
 	}
+
+	public function testSaveNewMessagePost_WhenSuccessful_TriggersHook() {
+		/* Arrange. */
+		$teacher_id = $this->factory->user->create( array( 'role' => 'teacher' ) );
+		$student_id = $this->factory->user->create( array( 'role' => 'subscriber' ) );
+		$course_id  = $this->factory->course->create();
+		$instance   = new Sensei_Messages();
+
+		// Remove the hooks to avoid side effects.
+		remove_all_actions( 'sensei_new_private_message' );
+
+		/* Act. */
+		$instance->save_new_message_post( $student_id, $teacher_id, 'message', $course_id );
+
+		/* Assert. */
+		$this->assertEquals( 1, did_action( 'sensei_new_private_message' ) );
+	}
+
+	public function testShowSuccessNotice_WhenNotRestRequest_DoesRedirect() {
+		/* Arrange. */
+		$instance = new Sensei_Messages();
+
+		$this->haltRedirects();
+
+		/* Act. */
+		try {
+			$instance->show_success_notice();
+		} catch ( \RuntimeException $e ) {
+			$redirect = json_decode( $e->getMessage(), true );
+		}
+
+		/* Assert. */
+		$this->assertSame( 302, $redirect['status'] );
+		$this->assertStringContainsString( 'send=complete', $redirect['location'] );
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
```

Comment:
This is needed to avoid polluting the global scope with the `REST_REQUEST` constant.